test/test_matgen.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2017-2023, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "slate/slate.hh"
+#include "test.hh"
+
+#include "scalapack_wrappers.hh"
+#include "scalapack_copy.hh"
+#include "print_matrix.hh"
+
+#include "matrix_utils.hh"
+#include "test_utils.hh"
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <utility>
+
+//------------------------------------------------------------------------------
+template <typename matrix_type>
+void test_matgen_work(Params& params, bool run)
+{
+    using scalar_t = typename matrix_type::value_type;
+    //using real_t = blas::real_type<scalar_t>;
+    //using blas::real;
+    //using blas::imag;
+    //using slate::ceildiv;
```

Comment:
I commented out some of these that are unused and the compiler might complain about. I also removed `alpha` and `beta`, which I don't think will be used for `matgen`.